reactors/1D_packbed.ipynb

```diff
@@ -0,0 +1,518 @@
+    "        drhoudz - sum_continuity - area2vol * membrane_flux * membrane_present\n",
+    "    )\n",
+    "\n",
+    "    # Conservation of gas-phase species\n",
+    "    for k in range(n_gas):\n",
+    "        res[offset_Y + k] = (\n",
+    "            dYdz[k]\n",
+    "            + (\n",
+    "                Y[k] * sum_continuity\n",
+    "                - phi * wdot_g[k] * W_g[k]\n",
+    "                - As * sdot_g[k] * W_g[k]\n",
+    "            )\n",
+    "            / rhou\n",
+    "        )\n",
+    "        if k == membrane_sp_ind:\n",
+    "            res[offset_Y + k] -= area2vol * membrane_flux * membrane_present\n",
+    "\n",
+    "    # Conservation of site fractions (algebraic contraints in this example)\n",
+    "    # for k in range(n_surf):\n",
+    "    res[offset_Z : offset_Z + n_surf] = sdot_s\n",
+    "\n",
+    "    # For the species with largest site coverage, use the constraint sum(Zk) = 1\n",
+    "    ind_large = np.argmax(coverages)\n",
+    "    res[offset_Z + ind_large] = 1 - np.sum(coverages)\n",
+    "\n",
+    "    # Conservation of momentum\n",
+    "    u = rhou / rho\n",
+    "    res[offset_p] = dpdz + phi * mu * u / B_g\n",
+    "\n",
+    "    # Conservation of energy\n",
+    "    conv_term = (4 / D_h) * h_coeff * (T_wall - T) * (2 * np.pi * R)\n",
+    "    chem_term = np.sum(hk_g * (phi * wdot_g + As * sdot_g))\n",
+    "\n",
+    "    res[offset_T] = dTdz  # isothermal condition\n",
+    "\n",
+    "    if solve_energy == True:\n",
+    "        res[offset_T] -= (conv_term - chem_term) / (rhou * cp)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Calculate the spatial derivatives at the inlet that will be used as the initial conditions for the IDA solver\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Initialize yPrime to 0 and call residual to get initial derivatives\n",
+    "y0 = np.hstack((rhou0, p_in, T_in, Yk_0, Zk_0))\n",
+    "yprime0 = np.zeros(n_var)\n",
+    "res = np.zeros(n_var)\n",
+    "residual(0, y0, yprime0, res)\n",
+    "for n in range(n_var):\n",
+    "    yprime0[n] = -res[n]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Solve the system of DAEs using ida solver"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "solver = dae(\n",
+    "    \"ida\",\n",
+    "    residual,\n",
+    "    first_step_size=1e-15,\n",
+    "    atol=1e-14,  # absolute tolerance for solution\n",
+    "    rtol=1e-06,  # relative tolerance for solution\n",
+    "    algebraic_vars_idx=[np.arange(offset_Y + n_gas, offset_Z + n_surf, 1)],\n",
+    "    max_steps=8000,\n",
+    "    one_step_compute=True,\n",
+    "    old_api=False,  # forces use of new api (namedtuple)\n",
+    ")\n",
+    "\n",
+    "time = []\n",
```

Comment:
I think you should call this list `distance`

Reply:
Updated.